<!DOCTYPE html>
<html class="no-js" lang="en">
<head>
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  <link href="http://cdnjs.cloudflare.com/ajax/libs/foundicons/3.0.0/foundation-icons.css" rel="stylesheet">
  <script type="text/javascript" async="" src="http://www.google-analytics.com/ga.js"></script>
  <meta class="foundation-data-attribute-namespace">
  <meta class="foundation-mq-xxlarge">
  <meta class="foundation-mq-xlarge-only">
  <meta class="foundation-mq-xlarge">
  <meta class="foundation-mq-large-only">
  <meta class="foundation-mq-large">
  <meta class="foundation-mq-medium-only">
  <meta class="foundation-mq-medium">
  <meta class="foundation-mq-small-only">
  <meta class="foundation-mq-small">
  <meta class="foundation-mq-topbar">
  <!-- If you delete this meta tag World War Z will become a reality -->
  <meta name="viewport" content="width=device-width, initial-scale=1.0">

  <title>MAnorm web manual, author:semal</title>
  <link rel="stylesheet" href="foundation-5/css/foundation.css">
  <link rel="stylesheet" href="foundation-5/css/foundation.min.css">
  <link rel="stylesheet" href="foundation-5/css/normalize.css">
  <script src="foundation-5/js/vendor/modernizr.js"></script>
  <style></style>

</head>

<body class="antialiased hide-extras">

<div class="contain-to-grid sticky">
  <nav class="top-bar" data-topbar role="navigation" data-options="sticky_on: large">
  <ul class="title-area">
    <li class="name">
      <h1><a href="http://bioinfo.sibs.ac.cn/shaolab/">ShaoLab</a></h1>
    </li>
  </ul>

  <section class="top-bar-section">
    <!-- Left Nav Section -->
    <ul class="left">
      <li class="has-dropdown"><a href="#section1">MAnorm Introduction</a>
        <ul class="dropdown">
          <li><a href="#1.1">Github and Paper</a></li>
          <li><a href="#1.2">Download and Install</a></li>
        </ul>
      </li>
      <li class="has-dropdown"><a href="#section2">MAnorm Help Info</a>
        <ul class="dropdown">
          <li><a href="#2.1">Options</a></li>
        </ul>
      <li>
    </ul>
  </section>
  </nav>
</div>

<section id="section1">
  <br>
  <br>
  <div class="row">
    <h2>MAnorm introduction </h2>
    <p>ChIP-Seq is widely used to characterize genome-wide binding patterns of transcription factors and other chromatin-associated proteins. Although comparison of ChIP-Seq data sets is critical for understanding cell type-dependent and cell state-specific binding, and thus the study of cell-specific gene regulation, few quantitative approaches have been developed. <br>Here, we present a simple and effective method, MAnorm, for quantitative comparison of ChIP-Seq data sets describing transcription factor binding sites and epigenetic modifications. The quantitative binding differences inferred by MAnorm showed strong correlation with both the changes in expression of target genes and the binding of cell type-specific regulators.</p>
    <hr>
    <h3 id="1.1">Github and Paper</h3>
    <p>If you want to get the source code of python version MAnorm, you should check out on github:<br><a href="https://github.com/semal/MAnormFast">MAnorm on Github</a></p>
    <p>And if you want to know the detail of this model, you could download the article:<br><a href="http://genomebiology.com/2012/13/3/R16/abstract">Shao Z, Zhang Y, Yuan GC, Orkin SH, Waxman DJ. (2012) MAnorm: a robust model for quantitative comparison of ChIP-Seq data sets. Genome Biol. Mar 16;13(3):R16.</a></p>
    <hr>
    <h3 id="1.2">Download and Install<h3>
    <p>After you download the compressed file, install it though these steps:</p>
    <p><a data-src-download="1.0.0" href="MAnormFast.tar.gz" title="MAnorm" class="button">Download MAnorm</a></p>
    <div class="row path-item">
      <ul>
      <li>Extract MAnormFast.tar.gz by typing on the command line:<br><code>$ tar xvf MAnormFast.tar.gz</code></li>
      <li>Enter the folder created by tar (i.e. ./MAnormFast) and type the following on the command line:<br>
        <code>$ cd MAnormFast</code><br><code>$ sudo python setup.py install</code></li>
      <ul>
    </div>
    <p class="subheader note"><strong>Note:</strong>the installer must have the proper security privilege to install MAnorm at /usr/local/bin; otherwise, the user can install MAnorm at the home as follows:<br><code>$ python setup.py install --prefix=$HOME</code><br>where $HOME stands for the user's home. Then, add a line to the log-in shell script (i.e. .bashrc in case of bash shell)<br><div class="panel"><p>export $PYTHONPATH=$PYTHONPATH:$HOME/lib/python2.7/site-packages<br>export PATH=pathOfMAnormFast/bin:$PATH</p></div></p>
  </div>
</section>

<section id="section2">
<div class="row">
  <hr>
  <h2>MAnormFast Help Info</h2>
  <p>After you install MAnormFast successfully, using <code>MAnormFast --help</code> for looking at the help information.<p>
  <hr>
  <h3 id="2.1">Options</h3>
  <p>There are lots of options used in MAnormFast commond, you could see them at help information. <br>And tables below also show each option to you.</p>
  <h4>5 required options<h4>
  <p>You could run MAnormFast using 5 options including the 4 input files and output folder name.<br><code>MAnormFast --p1 file1_path --p2 file2_path --r1 file3_path --r2 file4_path -o output_folder_name</code></p>
  <table>
    <thead>
      <tr>
        <th width="50">option</th>
        <th width="50">required</th>
        <th width="150">what represent</th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>--p1</td>
        <td>True</td>
        <td>peak file of sample 1, 3-col tab split, 4-col tab split bed like format and MACS peak file support</td>
      </tr>
      <tr>
        <td>--p2</td>
        <td>True</td>
        <td>Tpeak file of sample 2, 3-col tab split, 4-col tab split bed like format and MACS peak file support</td>
      </tr>
      <tr>
        <td>--r1</td>
        <td>True</td>
        <td>reads file of sample 1, only bed format support</td>
      </tr>
      <tr>
        <td>--r2</td>
        <td>True</td>
        <td>reads file of samaple 2, only bed format support</td>
      </tr>
      <tr>
        <td>-o</td>
        <td>True</td>
        <td>output result into a folder named by this option</td>
      </tr>
    </tbody>
  </table>
  <h4>Other Options</h4>
  <table>
    <thead>
      <tr>
        <th width="50">option</th>
        <th width="50">required</th>
        <th width="150">what represent</th>
        <th width="100">default value</th>
      </tr>
    </thead>
    <tr>
      <td>--s1</td>
      <td>False</td>
      <td>read shift size of sample 1, which should be set as the average size of DNA fragments.</td>
      <td>100</td>
    </tr>
    <tr>
      <td>--s2</td>
      <td>False</td>
      <td>read shift size of sample 2, which should be set as the average size of DNA fragments.</td>
      <td>100</td>
    </tr>
    <tr>
      <td>-n</td>
      <td>False</td>
      <td>number of random permutations to test the enrichment of overlapping between two peak sets.</td>
      <td>5</td>
    </tr>
    <tr>
      <td>-e</td>
      <td>False</td>
      <td>2*extension=size of the window centered at peak summit to calculate reads density. The window size should match the typical length of peaks, thus we recommend extension=1000 for sharp histone marks like H3K4me2/3 or H3K9/27ac, extension=500 for transcription factor or DNase-seq.</td>
      <td>1000</td>
    </tr>
    <tr>
      <td>-d</td>
      <td>False</td>
      <td>summit to summit distance cutoff, default=extension/2. Only those common peaks with distance between their summits in 2 samples smaller than this value will be considered as real common peaks for building the normalization model.</td>
      <td>extention/2</td>
    </tr>
    <tr>
      <td>-p</td>
      <td>False</td>
      <td>Cutoff of P-value to define biased (high-confidence sample 1 or 2-specific) peaks.</td>
      <td>0.01</td>
    </tr>
    <tr>
      <td>-m</td>
      <td>False</td>
      <td>Cutoff of M-value to define biased peaks. Sample 1 biased peaks are defined as sample 1 unique peaks with M-value > mcut_biased and P-value &lt; pcut_biased, while sample 2 biased peaks are defined as sample 2 unique peaks with M-value &lt; -1*mcut_biased and P-value &lt; pcut_biased.</td>
      <td>1</td>
    </tr>
    <tr>
      <td>-u</td>
      <td>False</td>
      <td>Cutoff of M-value to define unbiased (high-confidence non-specific) peaksbetween 2 samples, default=1. They are defined to be the common peaks with -1*mcut_unbiased &lt; M-value &lt; mcut_unbiased and P-value &gt; pcut_biased.</td>
      <td>1</td>
    </tr>
  </table>
  <h3>Other output related options<h3>
    <p>There are two options could chosed for setting output files.</p>
    <table>
      <thead>
        <tr>
          <td>option</td>
          <td>what represent</td>
        </tr>
      </thead>
      <tr>
        <td>-s</td>
        <td>By default, MAnorm will first separate both sets of input peaks into common and unique peaks, by checking whether they have overlap with any peak in the other sample, and then merge the 2 sets of common peaks into 1 group of non-overlapping ones. But if this option is used, MAnorm would not merge the common peaks and the peaks in output files will be exactly the same as those from input.</td>
      </tr>
      <tr>
        <td>-v</td>
        <td>if this option is used, MAnorm will choose biased peaks only from unique peaks, and choose unbiased peaks only from common peaks. But if this option is not used, MAnorm will choose biased and unbiased peaks just based on the M-value and P-value cutoffs, without checking whether they are common or unique peaks.</td>
      </tr>
    </table>

    <h3>Peak file format support</h3>
    <p>standard bed format is supported. and there are 3 other kinds of peak file format supported by MAnormFast, 3-col tab split format, 4-col tab split format and MACS peak file. 3-col including peak chromosome, peak start and peak end.</p>
    <ul class="tabs" data-tab role="tablist">
      <li class="tab-title active" role="presentation"><a href="#panel2-1" role="tab" tabindex="0" aria-selected="true" aria-controls="panel2-1">3-col tab split format</a></li>
      <li class="tab-title" role="presentation"><a href="#panel2-2" role="tab" tabindex="0" aria-selected="false" aria-controls="panel2-2">4-col tab split format</a></li>
      <li class="tab-title" role="presentation"><a href="#panel2-3" role="tab" tabindex="0" aria-selected="false" aria-controls="panel2-3">MACS peak file</a></li>
      <!-- <li class="tab-title" role="presentation"><a href="#panel2-4" role="tab" tabindex="0" aria-selected="false" aria-controls="panel2-4">Tab 4</a></li> -->
    </ul>
    <div class="tabs-content">
      <section role="tabpanel" aria-hidden="false" class="content active" id="panel2-1">
        <pre>
  chr1  2345  4345
  chr1  3456  5456
  chr2  6543  8543
        </pre>
      </section>
      <section role="tabpanel" aria-hidden="true" class="content" id="panel2-2">
        <pre>
  chr1  2345  4345  254
  chr1  3456  5456  127
  chr2  6543  8543  302

  note: the fourth column is summit distance from the start.
        </pre>
      </section>
      <section role="tabpanel" aria-hidden="true" class="content" id="panel2-3">
        <pre>
# This file is generated by MACS
# ARGUMENTS LIST:
# name = A549_H3K27acEtoh02_Rep1
# format = BED
# ChIP-seq file = /mnt/MAmotif/1.RAWdata/Histone_Broad_hg19/H3K27ac/wgEncodeBroadHistoneA549H3k27acEtoh02AlnRep1.bed
# control file = /mnt/MAmotif/1.RAWdata/Histone_Broad_hg19/modified_control/wgEncodeBroadHistoneA549ControlEtoh02AlnRep2.bed
# effective genome size = 3.14e+09
# tag size = 37
# band width = 300
# model fold = 32
# pvalue cutoff = 1.00e-05
# Ranges for calculating regional lambda are : peak_region,1000,5000,10000
# unique tags in treatment: 28191481
# total tags in treatment: 32226999
# unique tags in control: 27551475
# total tags in control: 32964797
# d = 200
chr     start   end     length  summit  tags    -10*log10(pvalue)       fold_enrichment FDR(%)
chr7    97911064        97917104        6041    368     648     3233.06 22.57   0.0
chr11   35159640        35167695        8056    1358    1066    3233.06 21.83   0.0
chr7    156684099       156687307       3209    2242    526     3233.06 43.37   0.0
chrX    55025266        55028474        3209    1270    551     3233.06 36.65   0.0
chr17   592847  601584  8738    4484    1286    3233.06 19.17   0.0
chr4    985500  988137  2638    1351    412     3233.06 34.69   0.0
        </pre>
      </section>
      <!-- <section role="tabpanel" aria-hidden="true" class="content" id="panel2-4">
        <h2>Fourth panel content goes here...</h2>
      </section> -->
      <hr>
      <p>2015 © Laboratory of Regulatory and Systems Genomics,PICB, SIBS, CAS</p>
    </div>
</div>
</section>

  <script src="foundation-5/js/vendor/jquery.js"></script>
  <script src="foundation-5/js/foundation.min.js"></script>
  <script>
    // $(document).foundation().foundation('joyride', 'start');
    $(document).foundation();
  </script>
</body>

</html>
